test/validation/api/scheduler/scheduler.c

```diff
@@ -415,10 +432,11 @@ static void scheduler_test_queue_size(void)
 				      ODP_SCHED_SYNC_ORDERED};
 
 	CU_ASSERT_FATAL(odp_queue_capability(&queue_capa) == 0);
+	CU_ASSERT_FATAL(odp_schedule_capability(&schedule_capa) == 0);
 	queue_size = TEST_QUEUE_SIZE_NUM_EV;
-	if (queue_capa.sched.max_size &&
-	    queue_size > queue_capa.sched.max_size)
-		queue_size = queue_capa.sched.max_size;
+	if (schedule_capa.max_queue_size &&
+	    queue_size > schedule_capa.max_queue_size)
+		queue_size = schedule_capa.max_queue_size;
```

Comment:
After config(NULL), this is a wrong assumption. Config(NULL) sets queue size to default. Since config() is shared over all validation test, easiest fix would be

odp_schedule_config_init(&default_config);

if (default_config.queue_size && queue_size > default_config.queue_size)
    queue_size = default_config.queue_size;

The same fix would be needed for all config(NULL) + comparison to capa.max_queue_size, but at least the validation test for scheduled queue size should be up to date.

Normally, application would do:
if (capa.max_queue_size && queue_size > capa.max_queue_size)
    queue_size = capa.max_queue_size;

config_init(&conf);
conf.queue_size = queue_size;
config(&conf);

Reply:
@psavol Currently scalable scheduler will set `max_queue_size` to 0. How should it be fixed?

Reply:
0 is valid value if queue size is "infinite" == only system memory / pool sizes limit. I'm not sure if scalable has infinite or finite size queues. Anyway, no need to fix it in this set.

Reply:
ok